src/Task.ts

```diff
@@ -862,11 +862,37 @@ export class Task {
         )})`;
     }
 
+    /*
+     * Escape a string so it can be used as part of a RegExp literally.
+     * Taken from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#escaping
+     */
+    private escapeRegExp(s: string) {
+        return s.replace(/([.*+?^=!:${}()|[]\/\\])/g, '\\$1');
+    }
+
+    /*
+     * Search for the global filter for the purpose of removing it from the description, but do so only
+     * if it is a separate word (preceding the beginning of line or a space and followed by the end of line
+     * or a space), because we don't want to cut-off nested tags like #task/subtag.
+     * If the global filter exists as part of a nested tag, we keep it untouched.
+     */
     public getDescriptionWithoutGlobalFilter() {
         const { globalFilter } = getSettings();
-        return this.description
-            .replace(globalFilter, '')
-            .replace('  ', ' ')
-            .trim();
+        let description = this.description;
+        // This matches the global filter (after escaping it) only when it's a complete word
+        const globalFilterRegex = RegExp(
+            '(^|\\s)' + this.escapeRegExp(globalFilter) + '($|\\s)',
+            'ug',
+        );
+        if (
+            globalFilter.length > 0 &&
```

Comment:
Could this be simplified by returning after 880 if the globalFilter is empty? Is there any value in creating the regex at all for an empty string?

Reply:
Done.